### PR TITLE
call finishInterrupt only when session has really been interrupted

### PR DESCRIPTION
**Pull request by jeffvroom**
### Intent

Fixes the "click interrupt three times to get the abort dialog. " part of #8865 after it was broken with async rpc. 

### Approach

With this PR, only finish the interrupt when the console busy is actually set to false

### Automated Tests

TBD - I could build an automated test to trigger the abort dialog if anyone thinks it's worth the effort

### QA Notes

Test the two ways to block the R console - both interruptible and non-interruptible and press the "Interrupt" button. 

Interruptible ops like system("sleep 30") should just return

Non-interruptible ops like the hang() example in #8865 should display the "abort R" dialog on the 3rd click in 10 seconds. 

### Checklist

- [x ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java [hunk 1]
```diff
@@ -186,6 +202,19 @@ public void onTerminateR()
...
+   @Override
+   public void onConsoleBusy(ConsoleBusyEvent event)
+   {
+      isBusy_ = event.isBusy();
+
+      if (!isBusy_ && pendingInterrupt_)
+      {
+         eventBus_.fireEvent(new InterruptStatusEvent(
+               InterruptStatusEvent.INTERRUPT_COMPLETED));
+         finishInterrupt(pendingInterruptHandler_);
```
Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/application/ApplicationInterrupt.java:215]: Should we reset `pendingInterrupt_` + `pendingInterruptHandler_` here (after invoking `finishInterrupt()`)?

**Review by kevinushey — COMMENTED** [review]
LGTM, but do we need to reset the "pending interrupt" state after we've handled a pending interrupt?

**Comment by kevinushey** [comment]
Does this also work in cases where a REPL in R is running? E.g. something like:

```
for (i in 1:10) {
    response <- tryCatch(
        readline("Enter a number: "),
        interrupt = function(e) { writeLines("Caught interrupt.") }
    )
}
```

What happens if you run that code and try to repeatedly interrupt it?

**Comment by jeffvroom** [comment]
Thanks, I'm super happy to have more test cases to try out. When I run this, I don't see the stop button available so I can't interrupt R when it's waiting for input. I see that the last event sent to the IDE is a busy=true so I'm guessing the IDE sees the console prompt event and turns off the stop button. Is that what we want or should you be able to interrupt it?  

Doesn't seem to be affected by these changes.

**Comment by kevinushey** [comment]
You should still be able to interrupt by pressing Escape or Ctrl + C, though -- does that work as expected? (I'm wondering if you can interrupt 10 times to exit that loop, or if you get the "R is not responding" warning on the 3rd interrupt attempt. Although, I could see arguments for either behavior being correct...)

**Comment by jeffvroom** [comment]
It does still let you interrupt 10 times to exit the loop. Took me a while to figure out why and a good excuse to figure out GWT's super dev mode :)  It's because interrupting it via a keypress doesn't use ApplicationInterrupt.java - it instead is called from Shell.java where there's just no logic to show the terminate dialog. It seems ok as it is to me, but I'm happy to make any changes to that other code.

**Comment by kevinushey** [comment]
Ah! Got it, that makes sense. Thanks for the extra investigation -- in that case, this LGTM!

**Pull request merged**